Firebase.CloudFirestore/source/Firebase.CloudFirestore/ApiDefinition.cs

```diff
@@ -0,0 +1,689 @@
+		void DeleteDocument ([NullAllowed] DocumentActionCompletionHandler completion);
+
+		// -(void)getDocumentWithCompletion:(FIRDocumentSnapshotBlock _Nonnull)completion;
+		[Async]
+		[Export ("getDocumentWithCompletion:")]
+		void GetDocument (DocumentSnapshotHandler completion);
+
+		// -(id<FIRListenerRegistration> _Nonnull)addSnapshotListener:(FIRDocumentSnapshotBlock _Nonnull)listener;
+		[Export ("addSnapshotListener:")]
+		IListenerRegistration AddSnapshotListener (DocumentSnapshotHandler listener);
+
+		// -(id<FIRListenerRegistration> _Nonnull)addSnapshotListenerWithOptions:(FIRDocumentListenOptions * _Nullable)options listener:(FIRDocumentSnapshotBlock _Nonnull)listener;
+		[Export ("addSnapshotListenerWithOptions:listener:")]
+		IListenerRegistration AddSnapshotListener ([NullAllowed] DocumentListenOptions options, DocumentSnapshotHandler listener);
+	}
+
+	// @interface FIRDocumentSnapshot : NSObject
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRDocumentSnapshot")]
+	interface DocumentSnapshot
+	{
+		// @property (readonly, assign, nonatomic) BOOL exists;
+		[Export ("exists")]
+		bool Exists { get; }
+
+		// @property (readonly, nonatomic, strong) FIRDocumentReference * _Nonnull reference;
+		[Export ("reference", ArgumentSemantic.Strong)]
+		DocumentReference Reference { get; }
+
+		// @property (readonly, copy, nonatomic) NSString * _Nonnull documentID;
+		[Export ("documentID")]
+		string Id { get; }
+
+		// @property (readonly, nonatomic, strong) FIRSnapshotMetadata * _Nonnull metadata;
+		[Export ("metadata", ArgumentSemantic.Strong)]
+		SnapshotMetadata Metadata { get; }
+
+		// -(NSDictionary<NSString *,id> * _Nonnull)data;
+		[Export ("data")]
+		NSDictionary<NSString, NSObject> Data { get; }
+
+		// -(id _Nullable)objectForKeyedSubscript:(id _Nonnull)key;
+		[return: NullAllowed]
+		[Export ("objectForKeyedSubscript:")]
+		NSObject GetObject (NSObject key);
+	}
+
+	// @interface FIRFieldPath : NSObject <NSCopying>
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRFieldPath")]
+	interface FieldPath : INSCopying
+	{
+		// -(instancetype _Nonnull)initWithFields:(NSArray<NSString *> * _Nonnull)fieldNames;
+		[Export ("initWithFields:")]
+		IntPtr Constructor (string [] fieldNames);
+
+		// +(instancetype _Nonnull)documentID;
+		[Static]
+		[Export ("documentID")]
+		FieldPath GetDocumentId ();
+	}
+
+	// @interface FIRFieldValue : NSObject
+	[DisableDefaultCtor]
+	[BaseType (typeof (NSObject), Name = "FIRFieldValue")]
+	interface FieldValue
+	{
+		// +(instancetype _Nonnull)fieldValueForDelete;
+		[Static]
+		[Export ("fieldValueForDelete")]
+		FieldValue Delete { get; }
+
+		// +(instancetype _Nonnull)fieldValueForServerTimestamp;
+		[Static]
+		[Export ("fieldValueForServerTimestamp")]
+		FieldValue ServerTimestamp { get; }
+	}
+
+	// id _Nullable (^)(FIRTransaction *, NSError **)
+	//delegate NSObject UpdateHandler (Transaction transaction, out NSError error);
```

Comment:
I think what you want here is not an `out` but a `ref` since I guess is up to you to set the error if any but not necessarily alway output it.